File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        char[] cArr = colors.toCharArray();
        int res = 0;
        for(int i = 1; i < cArr.length; ++i){
            if(cArr[i] == cArr[i-1]){
                res += Math.min(neededTime[i], neededTime[i-1]);
                neededTime[i] = Math.max(neededTime[i], neededTime[i-1]);
            }
        }
        
        return res;
    }
}

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
import heapq
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        res = 0
        min_heap = []
        for i in range(len(colors) - 1):
            if colors[i] != colors[i+1]:
                heapq.heappush(min_heap, neededTime[i])
                while len(min_heap) > 1:
                    res += heapq.heappop(min_heap)
                min_heap = []
            else:
                heapq.heappush(min_heap, neededTime[i])
        
        heapq.heappush(min_heap, neededTime[-1])

        while len(min_heap) > 1:
            res += heapq.heappop(min_heap)
        return res
                
            
            
            